### ISSUE TITLE
Users can't create a new account using LinkedIn.

### ISSUE THREAD

**Issue by HALVAM** [akeeba/sociallogin#99]
Hello, I'm having an issue with the LinkedIn integration plugin, for some reason it isn't allowing new users to create an account with LinkedIn, even though all related options are enabled.

After some testings, I have noticed that users with accents in their name like Rendón or Martínez are getting this issue.

It only affects registration with LinkedIn, users are able to create an account with their Facebook and Google accounts.

### Steps to reproduce the issue

- Configure LinkedIn integration so it can create new user accounts.
![image](https://github.com/akeeba/sociallogin/assets/23493800/e72d58e6-0fad-4d1f-8a1e-c928cdf19935)

- Add a login module on the frontpage that shows Social Login buttons.
- User click on the LinkedIn button and he is redirected to LinkedIn.
- He completes the verification process with LN and is send back to the frontpage.

### Expected result

The user is redirected to the frontpage and his new user account has been created.

### Actual result

- The account is not created, instead an error appear that says "You do not have an account on this site that corresponds to this LinkedIn account."

### Troubleshooting already performed

- Uninstall and reinstall plugin
- Compare LN integration settings against those of Google and Facebook.
- Enable Joomla's debug mode and setting error reporting to maximum.
- Review server logs.
- I'm currently disabling plugins and components, but at the moment there hasn't been any improvements.

#### Mandatory information

Issues without this information can not and _will not_ be replied to.

* **Joomla! version**: 4.3.2
* **PHP version**: 8.0.29
* **Am I using HTTPS**: Yes
* **Server Operating System**: Linux
* **My device's Operating System**: Windows
* **My device's Operating System version**: 10
* **Browser**: Vivald and Firefox
* **Browser version**: 6.1.3035.50 and 114.0.1

#### Good to have information

You can skip some or all of this information. However, the more information you provide the faster and better we can help.

* **MySQL version**: 10.6.12
* **Server type**: Apache
* **Server version**: N/A
* **Hosting company / Local server**: N/A
* **Server Operating System version**: N/A. 

### Additional comments

------

**Comment by nikosdion** [comment]
Every user account needs a username. The username is created by taking the full name, replacing the non-word characters with dots, and converting the string to lowercase.

I have tested this with users who have Greek characters in their name and it works just fine. I do not see a reason why Joomla would reject yours. The filter regular expression for usernames is `/[\x00-\x1F\x7F<>"\'%&]/` (in Joomla's \Joomla\Filter\InputFilter::cleanUsername) which allows all printable characters (and some special characters we are guaranteed not to use anyway).

Are you absolutely sure that no user was created? If you are 100% sure about that (please check!) then your problem must be a database collation one, e.g. your site's tables use a latin1 collation instead of the utf8mb4_general_ci collation Joomla normally uses. The former does not allow accented latin characters or anything that's not an ASCII character. The latter allows anything.

Otherwise, if the user has been created but not enabled, that's because the user has not verified their email with LinkedIn. When they do they become "verified" as far as SocialLogin knows and the user accounts created by SocialLogin will be activated.

**Comment by HALVAM** [comment]
Hello @nikosdion,

This past weekend, I conducted some tests and disabled even more components and plugins. Eventually, I found the culprit: a component called Joomdle, which facilitates the connection between Joomla's and Moodle's databases and content.

When a user is created in Joomla, the data is sent to Moodle. However, Moodle does not accept usernames with accents or similar special characters, which prevents the account from being created in both systems.

I modified src\Library\Plugin\LoginTrait.php by adding the following code inside the createUser function:

`$newname = iconv('UTF-8', 'ASCII//TRANSLIT', $name);`

After making these changes, Social Login is now able to register users normally.
